@jorisvandenbossche

```
=========================== short test summary info ============================
FAILED pandas/tests/io/test_user_agent.py::test_server_and_default_headers[ParquetPyArrowUserAgentResponder-read_parquet-34268-pyarrow]
FAILED pandas/tests/io/test_user_agent.py::test_server_and_custom_headers[JSONUserAgentResponder-read_json-34264-None]
FAILED pandas/tests/io/test_user_agent.py::test_server_and_custom_headers[ParquetPyArrowUserAgentResponder-read_parquet-34270-pyarrow]
FAILED pandas/tests/io/test_user_agent.py::test_server_and_custom_headers[GzippedJSONUserAgentResponder-read_json-34266-None]
= 4 failed, 136679 passed, 14311 skipped, 1054 xfailed, 63 xpassed in 1148.83s (0:19:08) =
```

--- a/ci/deps/actions-37-locale.yaml
+++ b/ci/deps/actions-37-locale.yaml
@@ -30,7 +30,7 @@ dependencies:
   - openpyxl
   - pandas-gbq
   - google-cloud-bigquery>=1.27.2 # GH 36436
-  - pyarrow>=0.17
+  - pyarrow=0.17 # GH 38803
   - pytables>=3.5.1
   - scipy
   - xarray=0.12.3

--- a/doc/source/whatsnew/v1.2.1.rst
+++ b/doc/source/whatsnew/v1.2.1.rst
@@ -32,6 +32,7 @@ I/O
 
 - Bumped minimum fastparquet version to 0.4.0 to avoid ``AttributeError`` from numba (:issue:`38344`)
 - Bumped minimum pymysql version to 0.8.1 to avoid test failures (:issue:`38344`)
+- Fixed ``AttributeError`` with PyArrow versions [0.16.0, 1.0.0) (:issue:`38801`)
 
 -
 -

--- a/pandas/core/arrays/string_arrow.py
+++ b/pandas/core/arrays/string_arrow.py
@@ -29,13 +29,12 @@ try:
 except ImportError:
     pa = None
 else:
-    # our min supported version of pyarrow, 0.15.1, does not have a compute
-    # module
-    try:
+    # PyArrow backed StringArrays are available starting at 1.0.0, but this
+    # file is imported from even if pyarrow is < 1.0.0, before pyarrow.compute
+    # and its compute functions existed. GH38801
+    if LooseVersion(pa.__version__) >= "1.0.0":
         import pyarrow.compute as pc
-    except ImportError:
-        pass
-    else:
+
         ARROW_CMP_FUNCS = {
             "eq": pc.equal,
             "ne": pc.not_equal,

--- a/pandas/tests/io/test_parquet.py
+++ b/pandas/tests/io/test_parquet.py
@@ -896,7 +896,7 @@ class TestParquetPyArrow(Base):
         # this use-case sets the resolution to 1 minute
         check_round_trip(df, pa, check_dtype=False)
 
-    @td.skip_if_no("pyarrow", min_version="0.17")
+    @td.skip_if_no("pyarrow", min_version="1.0.0")
     def test_filter_row_groups(self, pa):
         # https://github.com/pandas-dev/pandas/issues/26551
         df = pd.DataFrame({"a": list(range(0, 3))})

